wurst/lib/StackNSplit.wurst

```diff
@@ -0,0 +1,241 @@
+    if whichEvent == EVENT_ITEM_CHARGED_ADDED
+        return eventAddedTrigger
+    else if whichEvent == EVENT_ITEM_CHARGES_REMOVED
+        return eventRemoveTrigger
+    return null
+
+/** Checks if unit inventory is full. */
+public function unit.isInventoryFull() returns boolean
+    let size = this.inventorySize()
+
+    for i = 0 to size-1
+        if this.itemInSlot(i) == null
+            return false
+    return true
+
+/** Checks if unit inventory is fully stacked and no charges can be added. */
+public function unit.isItemFullyStacked(int itemTypeId) returns boolean
+    let max = getItemMaxStacks(itemTypeId)
+    item itm
+
+    if max > 0
+        let size = this.inventorySize()
+        for i = 0 to size-1
+            itm = this.itemInSlot(i)
+            if itm.getTypeId() == itemTypeId and GetItemCharges(itm) < max
+                return false
+    return true
+
+function fireEvent(trigger evt, unit u, item itm, int charges)
+    unit prevUnit = eventUnit
+    item prevItem = eventItem
+    integer prevCharges = eventCharges
+
+    eventUnit = u
+    eventItem = itm
+    eventCharges = charges
+
+    evt.evaluate()
+
+    eventUnit = prevUnit
+    eventItem = prevItem
+    eventCharges = prevCharges
+
+/** Returns value indicating whether specified item is registered as stackable. */
+public function isItemStackable(int itemTypeId) returns boolean
+    return stackTable.hasInt(itemTypeId)
+
+/** Registers specified item as stackable. */
+public function makeItemStackable(int itemTypeId, int stacks, int splits)
+    int value = splits
+
+    if stacks > 0
+        if value < 1
+            value = 1
+        stackTable.saveInt(itemTypeId, stacks)
+        stackTable.saveInt(-itemTypeId, value)
+
+/** Unregisters specified item from being stackable. */
+public function makeItemUnstackable(int itemTypeId)
+    stackTable.removeInt(itemTypeId)
+    stackTable.removeInt(-itemTypeId)
+
+/** Retrieves maximum amount of stacks for specified item. */
+public function getItemMaxStacks(int itemTypeId) returns int
+    if isItemStackable(itemTypeId)
+        return stackTable.loadInt(itemTypeId)
+    return 0
+
+/** Retrieves number of stacks are dropped when splitting. This number is not final and can change during runtime
+    if event item does not have required number of charges. Capped at minimum of 1. */
+public function getItemSplitCount(int itemTypeId) returns int
+    if isItemStackable(itemTypeId)
+        return stackTable.loadInt(-itemTypeId)
+    return 0
+
+/** Attempts to stack provided item for specified unit. */
+public function stackItem(unit u, item itm) returns boolean
+    int charges = GetItemCharges(itm)
+    int itemTypeId = itm.getTypeId()
+    int curr
+    int diff
+    int size = u.inventorySize()
+    item with
+
+    if charges == 0 or not isItemStackable(itemTypeId)
+        return false
+
+    int max = getItemMaxStacks(itemTypeId)
+    for i = 0 to size-1
```

Comment:
I don't think it makes sense to omit spaces around binary operators in control statements but not assignments, e.g. `diff = max - curr`. Not too concerned with this though.

Reply:
Thanks!